server/cluster/cluster.go

```diff
@@ -444,6 +446,39 @@ func (c *RaftCluster) RemoveSuspectRegion(id uint64) {
 	c.suspectRegions.Remove(id)
 }
 
+// AddSuspectKeyRanges add the key ranges with the its ruleID as the key
+// The instance of each keyRange is like following format:
+// [2][]byte: start key/ned key
+func (c *RaftCluster) AddSuspectKeyRanges(key string, keyRanges [2][]byte) {
+	c.Lock()
+	defer c.Unlock()
+	c.suspectKeyRanges.Put(key, keyRanges)
+}
+
+// PopOneSuspectKeyRange get one suspect keyRange group
+// return value and true if pop success, return empty [][2][]byte and false
+// if suspectKeyRanges couldn't pop keyRange group.
+func (c *RaftCluster) PopOneSuspectKeyRange() (string, [2][]byte, bool) {
+	c.Lock()
+	defer c.Unlock()
+	key, value, success := c.suspectKeyRanges.Pop()
+	if !success {
+		return "", [2][]byte{}, false
+	}
+	v, ok := value.([2][]byte)
+	if !ok {
+		return "", [2][]byte{}, false
```

Comment:
looks impossible

Reply:
Yes, but I still suggest remain this logic as we can't ensure the value type.